detect.go

```diff
@@ -18,6 +20,10 @@ func Detect(env Environment) packit.DetectFunc {
 
 		_, packageManager, err := ScriptsToRun(filepath.Join(context.WorkingDir, env.NodeProjectPath), env.NodeRunScripts)
 		if err != nil {
+			if errors.Is(err, os.ErrNotExist) {
+				return packit.DetectResult{}, packit.Fail
```

Comment:
Can we include an extra message explaining why detection failed? This is especially helpful when trying to debug detection issues.

```suggestion
				return packit.DetectResult{}, packit.Fail.WithMessage("no package.json file present")
```

Reply:
good point, the PR has been updated

Reply:
You'll need to update the test to match.